app/reducers/articleReducers.js

```diff
@@ -10,14 +12,37 @@ const initialState = {
   highlights: [],
 }
 
+function mergeHighlights(list) {
+  // TODO: write tests for me
```

Comment:
just reproduced the bug. it's: highlight two separate pieces of text. start a new highlight inside the first highlight, and end it inside the second highlight. it can be fixed by highlighting over the resulting buggy region.